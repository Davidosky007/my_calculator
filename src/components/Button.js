/* eslint-disable linebreak-style */
import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ name, handleClick, className }) => (
  <div>
    <button type="button" onClick={() => handleClick(name)} className={className}>
      { name }
      {' '}
    </button>
  </div>
);

Button.propTypes = {
  name: PropTypes.string,
  handleClick: PropTypes.func,
  className: PropTypes.string,
};

Button.defaultProps = {
  name: null,
  handleClick: null,
  className: 'btn button',
};

export default Button;
