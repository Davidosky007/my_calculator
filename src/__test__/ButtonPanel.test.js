/* eslint-disable linebreak-style */
import { create } from 'react-test-renderer';
import '@testing-library/jest-dom';
import ButtonPannel from '../components/ButtonPanel';
import Button from '../components/Button';

describe('ButtonPanel component test', () => {
  test('render ButtonPanel component properly', () => {
    const buttonPanel = create(<ButtonPannel />).toJSON();
    expect(buttonPanel).toMatchSnapshot();
  });
  test('render Button component properly', () => {
    const button = create(<Button />).toJSON();
    expect(button).toMatchSnapshot();
  });
});
