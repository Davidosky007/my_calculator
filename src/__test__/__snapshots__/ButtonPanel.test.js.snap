// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPanel component test render Button component properly 1`] = `
<button
  className="btn button"
  onClick={[Function]}
  type="button"
>
   
</button>
`;

exports[`ButtonPanel component test render ButtonPanel component properly 1`] = `
<div
  className="btns"
>
  <div
    className="btn_grp"
  >
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      AC
       
    </button>
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      +/-
       
    </button>
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      %
       
    </button>
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      รท
       
    </button>
  </div>
  <div
    className="btn_grp"
  >
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      7
       
    </button>
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      8
       
    </button>
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      9
       
    </button>
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      x
       
    </button>
  </div>
  <div
    className="btn_grp"
  >
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      4
       
    </button>
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      5
       
    </button>
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      6
       
    </button>
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      -
       
    </button>
  </div>
  <div
    className="btn_grp"
  >
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      1
       
    </button>
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      2
       
    </button>
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      3
       
    </button>
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      +
       
    </button>
  </div>
  <div
    className="btn_grp2"
  >
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      0
       
    </button>
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      .
       
    </button>
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      =
       
    </button>
  </div>
</div>
`;
